solve1/main.cpp

```diff
@@ -0,0 +1,413 @@
+}
+
+/*
+@brief 根据匹配的矩形列表找到装甲板框，匹配方法是以两个灯条的中心点，根据fac竖直划两条线，之后连成一个矩形
+@param match_list：匹配的矩形列表
+@return 装甲板的矩形列表
+*/
+RectAreaList getPlateList(const RectAreaList &match_list)
+{
+    //左右两边化竖线的长度扩大比例，线的长度为2*fac*灯条长度
+    constexpr double fac = 1.35;
+    RectAreaList plate_list;
+    for (size_t i = 0; i < match_list.size(); i += 2)
+    {
+        std::vector<cv::Point2f> tmp_ve;
+        double p1_max_value = fac * std::max(pointDis(match_list[i][0], match_list[i][1]), pointDis(match_list[i][1], match_list[i][2]));
+        double p2_max_value = fac * std::max(pointDis(match_list[i + 1][0], match_list[i + 1][1]), pointDis(match_list[i + 1][1], match_list[i + 1][2]));
+        auto p1 = rectCenter(match_list[i]);
+        auto p2 = rectCenter(match_list[i + 1]);
+        tmp_ve.push_back(cv::Point2f(p1.x, p1.y - p1_max_value));
+        tmp_ve.push_back(cv::Point2f(p2.x, p2.y - p2_max_value));
+        tmp_ve.push_back(cv::Point2f(p2.x, p2.y + p2_max_value));
+        tmp_ve.push_back(cv::Point2f(p1.x, p1.y + p1_max_value));
+        plate_list.push_back(tmp_ve);
+    }
+    return plate_list;
+}
+
+/*
+@brief 单纯的在一个黑幕上绘制矩形框，为了展示方便
+@param name：显示窗口的名称
+@param gray_img：灰度图，为了生成黑幕图片
+@param rect_list：矩形框列表
+@param size：窗口尺寸
+*/
+void showDemo(const cv::String name, cv::Mat &gray_img, RectAreaList &rect_list, const cv::Size &size)
+{
+    cv::Mat &&demo = grayToBin(gray_img, 255);
+    drawRectInImg(demo, rect_list, cv::Scalar(255, 255, 255), 2);
+    showImg(name, demo, size);
+}
+
+int main()
+{
+    //图片路径
+    const cv::String img_get = "test.jpg";
+    //窗口大小，根据给定图片设置了3：2大小
+    constexpr int img_width = 900;
+    const cv::Size img_size(img_width, img_width / 3 * 2);
+    //灯条二值化阈值,tv：threshold_value
+    constexpr int tv_light = 215;
+    //数字二值化阈值
+    constexpr int tv_num = 155;
+    //画框的颜色和粗细
+    const auto rect_color = cv::Scalar(242, 250, 247);
+    const int rect_thickness = 10;
+    //输出路径
+    const cv::String img_out = "armor.jpg";
+
+    //读取图片并且进行判断,返回右值，减少拷贝
+    cv::Mat &&img = readImg(img_get);
+
+    //获取灰度图片
+    cv::Mat &&gray_img = normalToGray(img);
+    //获取灯条二值化图片
+    cv::Mat &&bin_img_light = grayToBin(gray_img, tv_light);
+    showImg("灯条二值化图片", bin_img_light, img_size);
+    //获取数字二值化图片
+    cv::Mat &&bin_img_num = grayToBin(gray_img, tv_num);
+    showImg("数字二值化图片", bin_img_num, img_size);
+
+    //获取灯条轮廓
+    RectAreaList &&rect_area = getRectArea(bin_img_light);
+    showDemo("灯条轮廓", gray_img, rect_area, img_size);
+
+    //筛选轮廓
+    RectAreaList &&selected_rect_area = selectRectArea(rect_area, selectMethod);
+    showDemo("筛选轮廓", gray_img, selected_rect_area, img_size);
+    //对轮廓进行排序
+    std::sort(selected_rect_area.begin(), selected_rect_area.end(), rectCmp);
+    //获得配对以后的矩形框
+    RectAreaList &&match_rect_list = matchRect(bin_img_num, selected_rect_area);
+    showDemo("灯条与配对结果", gray_img, match_rect_list, img_size);
+    //获得装甲板矩形框
+    RectAreaList &&plate_list = getPlateList(match_rect_list);
```

Comment:
maybe you can define  class `armor` and class  `light` to record these information.